### PR TITLE
21.0.0.10-beta blog post - Draft

### PR DESCRIPTION
**Pull request by ryan-storey**

### CODE REVIEW

File: posts/2021-09-03-210010-beta.adoc [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    <groupId>org.eclipse.microprofile.context-propagation</groupId>
+    <artifactId>microprofile-context-propagation-api</artifactId>
+    <version>1.3-RC1</version>
+</dependency>
+----
+
+For more information on this MicroProfile Context Propagation update, refer to the link:https://download.eclipse.org/microprofile/microprofile-context-propagation-1.3-RC1/apidocs/[JavaDoc] and link:https://download.eclipse.org/microprofile/microprofile-context-propagation-1.3-RC1/microprofile-context-propagation-spec-1.3-RC1.html[specification].
+
+[#value_add]
+=== Jakarta EE 9 support for the remaining Open Liberty value-add features
```
Comment (mbroz2) [inline posts/2021-09-03-210010-beta.adoc]: @jhanders34 does this complete our EE9 Open Liberty features, or are there still outstanding features that need converting/support?  If we're done, we should state that.

File: posts/2021-09-03-210010-beta.adoc [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+If you try either package, <<feedback, let us know what you think>>.
+[#allbeta]
+== All Beta Features package
+
+The All Beta Features package includes the following beta features:
+
+* <<java17, Java 17 Beta Support>>
+* <<mp_cp, MicroProfile Context Propagation 1.3>>
+* <<value_add, Jakarta EE 9 support for the remaining Open Liberty value-add features>>
+* <<jwe, JWE support for OpenID Connect Client and Social Media Login>>
```
Comment (mbroz2) [inline posts/2021-09-03-210010-beta.adoc]: JWE should be the second item in the blog, so move it up here and the actual section below to follow java17.

File: posts/2021-09-03-210010-beta.adoc [hunk 3]
```diff
@@ -0,0 +1,237 @@
+---
+layout: post
+title: "Java 17 beta support, MicroProfile Context Propagation 1.3, and other enhancements in Open Liberty 21.0.0.10-beta"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/ryan-storey
+author_github: https://github.com/ryan-storey
+seo-title: Java 17 beta support, MicroProfile Context Propagation 1.3, and other enhancements in Open Liberty 21.0.0.10-beta - OpenLiberty.io
```
Comment (mbroz2) [inline posts/2021-09-03-210010-beta.adoc]: Java 17 beta support, JWE support for access and ID tokens, MicroProfile Context Propagation 1.3, and other enhancements in Open Liberty 21.0.0.10-beta - OpenLiberty.io

File: posts/2021-09-03-210010-beta.adoc [hunk 4]
```diff
@@ -0,0 +1,237 @@
+---
+layout: post
+title: "Java 17 beta support, MicroProfile Context Propagation 1.3, and other enhancements in Open Liberty 21.0.0.10-beta"
```
Comment (mbroz2) [inline posts/2021-09-03-210010-beta.adoc]: Java 17 beta support, JWE support for access and ID tokens, MicroProfile Context Propagation 1.3, and other enhancements in Open Liberty 21.0.0.10-beta

File: posts/2021-09-03-210010-beta.adoc [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/ryan-storey
+author_github: https://github.com/ryan-storey
+seo-title: Java 17 beta support, MicroProfile Context Propagation 1.3, and other enhancements in Open Liberty 21.0.0.10-beta - OpenLiberty.io
+seo-description: Open Liberty 21.0.0.10-beta provides long-awaited Java 17 beta support, offering a lot of new enhancements and functionality. Also included are updates to MicroProfile Context Propagation, meaning that you can now use MicroProfile Context Propagation within your Jakarta EE 9 applications. This beta also offers JWE token format support for OpenID Connect Client and Social Media Login, as well as Jakarta EE 9 support for the remaining Open Liberty value-add features.
```
Comment (mbroz2) [inline posts/2021-09-03-210010-beta.adoc]: Open Liberty 21.0.0.10-beta provides long-awaited Java 17 beta support, offering a lot of new enhancements and functionality. Also included is JWE token format support for OpenID Connect Client (OIDC) and Social Media Login. This beta also offers updates to MicroProfile Context Propagation, meaning that you can now use MicroProfile Context Propagation within your Jakarta EE 9 applications, as well as Jakarta EE 9 support for the remaining Open Liberty value-add features.

File: posts/2021-09-03-210010-beta.adoc [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+blog_description: Open Liberty 21.0.0.10-beta provides long-awaited Java 17 beta support, offering a lot of new enhancements and functionality. Also included are updates to MicroProfile Context Propagation, meaning that you can now use MicroProfile Context Propagation within your Jakarta EE 9 applications. This beta also offers JWE token format support for OpenID Connect Client and Social Media Login, as well as Jakarta EE 9 support for the remaining Open Liberty value-add features.
```
Comment (mbroz2) [inline posts/2021-09-03-210010-beta.adoc]: Open Liberty 21.0.0.10-beta provides long-awaited Java 17 beta support, offering a lot of new enhancements and functionality. Also included is JWE token format support for OpenID Connect Client (OIDC) and Social Media Login. This beta also offers updates to MicroProfile Context Propagation, meaning that you can now use MicroProfile Context Propagation within your Jakarta EE 9 applications, as well as Jakarta EE 9 support for the remaining Open Liberty value-add features.

File: posts/2021-09-03-210010-beta.adoc [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Java 17 beta support, MicroProfile Context Propagation 1.3, and other enhancements in Open Liberty 21.0.0.10-beta
+Ryan Storey <https://github.com/Ryan Storey>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Open Liberty 21.0.0.10-beta provides long-awaited Java 17 beta support, offering a lot of new enhancements and functionality. Also included are updates to MicroProfile Context Propagation, meaning that you can now use MicroProfile Context Propagation within your Jakarta EE 9 applications. This beta also offers JWE token format support for OpenID Connect Client and Social Media Login, as well as Jakarta EE 9 support for the remaining Open Liberty value-add features.
```
Comment (mbroz2) [inline posts/2021-09-03-210010-beta.adoc]: Open Liberty 21.0.0.10-beta provides long-awaited Java 17 beta support, offering a lot of new enhancements and functionality. Also included is JWE token format support for OpenID Connect Client (OIDC) and Social Media Login. This beta also offers updates to MicroProfile Context Propagation, meaning that you can now use MicroProfile Context Propagation within your Jakarta EE 9 applications, as well as Jakarta EE 9 support for the remaining Open Liberty value-add features.

File: posts/2021-09-03-210010-beta.adoc [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+----
+
+For more information on this MicroProfile Context Propagation update, refer to the link:https://download.eclipse.org/microprofile/microprofile-context-propagation-1.3-RC1/apidocs/[JavaDoc] and link:https://download.eclipse.org/microprofile/microprofile-context-propagation-1.3-RC1/microprofile-context-propagation-spec-1.3-RC1.html[specification].
+
+[#value_add]
+=== Jakarta EE 9 support for the remaining Open Liberty value-add features
+
+In Open Liberty 21.0.0.10-beta we now have Jakarta EE 9 support for the following value-add features:
+
+* Admin Center (adminCenter-1.0)
```
Comment (mbroz2) [inline posts/2021-09-03-210010-beta.adoc]: make each of these link to their corresponding feature in docs. For example 
Admin Center ([adminCenter-1.0]({url-prefix}/docs/latest/reference/feature/adminCenter-1.0.html))

File: posts/2021-09-03-210010-beta.adoc [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+
+By taking advantage of trying out the new changes in Java 17 now, you get more time to review your applications, microservices and runtime environments and be a step ahead when it becomes generally available.
+
+To give this a try today, download the link:https://jdk.java.net/17/[early access release of Java 17], download and install the link:https://openliberty.io/downloads/#runtime_betas[21.0.0.10-beta] version of Open Liberty, edit your Liberty server's link:https://openliberty.io/docs/21.0.0.8/reference/config/server-configuration-overview.html#server-env[server.env file] to point JAVA_HOME to your Java 17 installation and start testing!
+
+For more information on Java 17, please visit the Java 17 link:https://jdk.java.net/17/release-notes[release notes page], link:https://download.java.net/java/early_access/jdk17/docs/api/[API Javadoc page] or link:https://jdk.java.net/17/[download page].
+For more information on Open Liberty, please visit our link:https://openliberty.io/docs/21.0.0.8/overview.html[documentation page].
+
+*Note: As we work towards full Java 17 support ourselves, please excuse any of our functionality that might not be 100% ready yet.
```
Comment (mbroz2) [inline posts/2021-09-03-210010-beta.adoc]: Add that we'd like to receive feedback for any issues they hit, something like:
`If you do hit any issues, let us know by providing <<feedback, feedback>>`

**Review by mbroz2 — CHANGES_REQUESTED** [review]
All /docs/ links that specify a version (like 21.0.0.8), need to be updated to use /latest/ instead.
All internal links need to be updated to use {url-prefix}.
For example, instead of `link:https://openliberty.io/docs/21.0.0.8/overview.html[documentation page]` you should have `link:{url-prefix}/docs/latest/overview.html[documentation page]`

File: posts/2021-09-03-210010-beta.adoc (hunk 1 above)
Comment (jhanders34) [reply posts/2021-09-03-210010-beta.adoc]: Yes all Open Liberty features that can and should support Jakarta EE 9 are enabled to work with Jakarta EE 9 in this beta.

File: posts/2021-09-07-java17-210010-beta.adoc [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+* link:https://openjdk.java.net/jeps/377[ZGC: A Scalable Low-Latency Garbage Collector]
+* link:https://openjdk.java.net/jeps/378[Text Blocks]
+* link:https://openjdk.java.net/jeps/380[Unix-Domain Socket Channels]
+* link:https://openjdk.java.net/jeps/390[Warnings for Value-Based Classes]
+* link:https://jdk.java.net/16/release-notes#JDK-8202343[Removal of TLS 1.0 and 1.1]
+* link:https://openjdk.java.net/jeps/394[Pattern Matching for instanceof]
+* link:https://openjdk.java.net/jeps/395[Records]
+* link:https://openjdk.java.net/jeps/403[Strongly Encapsulate JDK Internals]
+* link:https://openjdk.java.net/jeps/409[Sealed Classes]
+* link:https://openjdk.java.net/jeps/411[Deprecate the Security Manager for Removal]
```
Comment (mbroz2) [inline posts/2021-09-07-java17-210010-beta.adoc:59]: * link:https://bugs.openjdk.java.net/browse/JDK-8196415[Disable SHA-1 JARs]
* link:https://bugs.openjdk.java.net/browse/JDK-8259709[Disable SHA-1 XML Signatures]

Comment (mbroz2) [reply posts/2021-09-07-java17-210010-beta.adoc:59]: add those two above

**Review by mbroz2 — CHANGES_REQUESTED** [review]

**Review by mbroz2 — CHANGES_REQUESTED** [review]
There are still links pointing to openliberty.io in the URL.  These need to be updated to use `{url-prefix}` instead, see comment from previous review:

> All `/docs/` links that specify a version (like 21.0.0.8), need to be updated to use `/latest/` instead.
> All internal links need to be updated to use `{url-prefix}`.
> For example, instead of `link:https://openliberty.io/docs/21.0.0.8/overview.html[documentation page]` you should have `link:{url-prefix}/docs/latest/overview.html[documentation page]`

**Review by mbroz2 — APPROVED** [review]

**Pull request merged**